src/ast/ast.c

```diff
@@ -504,34 +497,50 @@ inline AST_AnnotationCtxCollection *AST_GetAnnotationCtxCollection(AST *ast) {
 	return ast->anot_ctx_collection;
 }
 
-static void _free_to_string_annotation
-(
-	void *userdata,
-	const cypher_astnode_t *node,
-	void *annotation
-) {
-	rm_free(annotation);
+// Compute the number of digits in a non-negative integer.
+static inline int _digit_count(int n) {
+	int count = 0;
+	do {
+		n /= 10;
+		count++;
+	} while(n);
+	return count;
 }
 
-char *AST_ToString(const cypher_astnode_t *node) {
+static inline char *_create_anon_alias(int anon_count) {
+	// We need space for "anon_" (5), all digits, and a NULL terminator (1)
+	int alias_len = _digit_count(anon_count) + 6;
+	char *alias = rm_malloc(alias_len * sizeof(char));
+	snprintf(alias, alias_len, "anon_%d", anon_count);
```

Comment:
remove `_digit_count`, replace `snprintf` with `asprintf` no need to `rm_malloc`